aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -233,7 +233,6 @@ def setup(self):
         internal loop.
         """
         super().setup()
-        self.ctx.restart_calc = None
```

Comment:
I'm not sure if I agree with this approach. I think it's fine to override inputs when restarting with a handler, but we should try and avoid overriding the initial inputs provided by the user, instead relying on their common sense and the validator. For the first iteration, I don't think we should call `set_restart_type`. For restarts, perhaps we should _always_ call `set_restart_type`, so there is no confusion or incorrect restart?

I do agree with switching to the `parent_folder` as an input instead of `calculation`, though, let me already make that change.